pkg/machinepool/machinepool.go

```diff
@@ -1228,25 +1228,81 @@ func appendUpgradesIfExist(scheduledUpgrade *cmv1.NodePoolUpgradePolicy, output
 	return output
 }
 
-func getMachinePoolsString(machinePools []*cmv1.MachinePool) string {
-	outputString := "ID\tAUTOSCALING\tREPLICAS\tINSTANCE TYPE\tLABELS\t\tTAINTS\t" +
-		"\tAVAILABILITY ZONES\t\tSUBNETS\t\tSPOT INSTANCES\tDISK SIZE\tSG IDs\n"
-	for _, machinePool := range machinePools {
-		outputString += fmt.Sprintf("%s\t%s\t%s\t%s\t%s\t\t%s\t\t%s\t\t%s\t\t%s\t%s\t%s\n",
-			machinePool.ID(),
-			ocmOutput.PrintMachinePoolAutoscaling(machinePool.Autoscaling()),
-			ocmOutput.PrintMachinePoolReplicas(machinePool.Autoscaling(), machinePool.Replicas()),
-			machinePool.InstanceType(),
-			ocmOutput.PrintLabels(machinePool.Labels()),
-			ocmOutput.PrintTaints(machinePool.Taints()),
-			output.PrintStringSlice(machinePool.AvailabilityZones()),
-			output.PrintStringSlice(machinePool.Subnets()),
-			ocmOutput.PrintMachinePoolSpot(machinePool),
-			ocmOutput.PrintMachinePoolDiskSize(machinePool),
-			output.PrintStringSlice(machinePool.AWS().AdditionalSecurityGroupIds()),
-		)
+func getMachinePoolsString(
+	r *rosa.Runtime,
+	machinePools []*cmv1.MachinePool,
+	showAZType, showDedicated, showWindowsLI, showAll bool,
+) string {
+	columnDefinitions := []struct {
+		header  string
+		extract func(*cmv1.MachinePool) string
+		enabled bool
```

Comment:
These feel really generic for names. I don't think `header` nor `generic` nor `enabled` are proper names for what this is accomplishing, maybe use ones more like what you named these values elsewhere/a bit more verbose;

`columnHeader`, `columnValue`, `visible`